pkg/platform/kube/functionres/lazy.go

```diff
@@ -2184,23 +2191,68 @@ func (lc *lazyClient) resolveFailFast(pods []v1.Pod,
 				}
 			}
 		}
+	}
 
+	errGroup, _ := errgroup.WithContext(context.TODO(), lc.logger)
+	for _, pod := range pods {
+		pod := pod
 		for _, condition := range pod.Status.Conditions {
 
 			// check if the pod is in pending state, and the reason is that it is unschedulable
 			// (meaning no k8s node can currently run it, because of insufficient resources etc..)
 			if pod.Status.Phase == v1.PodPending && condition.Reason == "Unschedulable" {
 
-				// TODO: if pods has an event that cluster is going to scale up - don't return error
-
-				return errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
+				errGroup.Go("WaitAndCheckAutoScaleEvents", func() error {
+					time.Sleep(15 * time.Second)
+
+					triggeredScaleUp, err := lc.isPodAutoScaledUp(pod, namespace)
+					if err != nil {
+						if triggeredScaleUp {
+							return errors.Wrap(err, "Failed to get pod events")
+						}
+						return errors.Wrapf(err,
+							"Autoscaler is unable to schedule NuclioFunction pod (%s)",
+							pod.Name)
+					}
+					return nil
+				})
 			}
 		}
 	}
-
+	if err := errGroup.Wait(); err != nil {
+		return errors.Wrap(err, "NuclioFunction pod is unavailable")
+	}
 	return nil
 }
 
+func (lc *lazyClient) isPodAutoScaledUp(pod v1.Pod, namespace string) (bool, error) {
+
+	// get pod events to check if pod triggered auto scale
+	//podEvents, err := lc.kubeClientSet.CoreV1().Events(namespace).List(metav1.ListOptions{
+	//	FieldSelector: fmt.Sprintf("involvedObject.name=%s", pod.Name),
+	//})
+	podEvents, err := lc.kubeClientSet.CoreV1().Events(namespace).List(metav1.ListOptions{})
+	if err != nil {
+
+		// true + err tell us there was an error getting events
+		return true, err
+	}
+
+	for _, event := range podEvents.Items {
+
+		if event.Source.Component == "cluster-autoscaler" {
+			if event.Reason == "TriggerScaleUp" {
+
+				// TODO: increase timeout?
+				return true, nil
+			} else if event.Reason == "NotTriggerScaleUp" || event.Reason == "ScaleDown" {
+				return false, errors.Errorf(strings.TrimSpace(event.Message))
+			}
+		}
+	}
+	return false, errors.Errorf("NuclioFunction pod (%s) is unschedulable", pod.Name)
```

Comment:
are you sure? that means, if pod have no events it is determined by you that it is unschedulable. 
I say, if a pod is running without events - it is still unknown and hence
`return false, nil` is more correct

Reply:
The entire function is called only when a pod is pending with and "Unschedulable" reason.
If we don't find an autoscale event - the pod is still unschedulable.